pkg/flowaggregator/exporter/ipfix_test.go

```diff
@@ -288,3 +255,104 @@ func TestIPFIXExporter_sendRecord_Error(t *testing.T) {
+	}
+	for _, ie := range antreaInfoElements {
+		elemList = append(elemList, createElement(ie, ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(ie, ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+	}
+	for i := range infoelements.StatsElementList {
+		elemList = append(elemList, createElement(infoelements.AntreaSourceStatsElementList[i], ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(infoelements.AntreaSourceStatsElementList[i], ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+		elemList = append(elemList, createElement(infoelements.AntreaDestinationStatsElementList[i], ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(infoelements.AntreaDestinationStatsElementList[i], ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+	}
+	for _, ie := range infoelements.AntreaFlowEndSecondsElementList {
+		elemList = append(elemList, createElement(ie, ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(ie, ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+	}
+	for i := range infoelements.AntreaThroughputElementList {
+		elemList = append(elemList, createElement(infoelements.AntreaThroughputElementList[i], ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(infoelements.AntreaThroughputElementList[i], ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+		elemList = append(elemList, createElement(infoelements.AntreaSourceThroughputElementList[i], ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(infoelements.AntreaSourceThroughputElementList[i], ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+		elemList = append(elemList, createElement(infoelements.AntreaDestinationThroughputElementList[i], ipfixregistry.AntreaEnterpriseID))
+		mockIPFIXRegistry.EXPECT().GetInfoElement(infoelements.AntreaDestinationThroughputElementList[i], ipfixregistry.AntreaEnterpriseID).Return(elemList[len(elemList)-1].GetInfoElement(), nil)
+	}
+	return elemList
+}
+
+func TestInitExportingProcess(t *testing.T) {
+	t.Run("tcp success", func(t *testing.T) {
+		k8sClientset := fake.NewSimpleClientset()
+		ctrl := gomock.NewController(t)
+		mockIPFIXRegistry := ipfixtesting.NewMockIPFIXRegistry(ctrl)
+		opt := &options.Options{}
+		opt.Config = &flowaggregatorconfig.FlowAggregatorConfig{}
+		flowaggregatorconfig.SetConfigDefaults(opt.Config)
+		listener, err := net.Listen("tcp", "127.0.0.1:0")
+		require.NoError(t, err)
+		defer listener.Close()
+		opt.ExternalFlowCollectorAddr = listener.Addr().String()
+		opt.ExternalFlowCollectorProto = listener.Addr().Network()
+		opt.Config.FlowCollector.RecordFormat = "JSON"
```

Comment:
why do this for TCP and not UDP?

Reply:
I actually wanted to have one case where RecordFormat is assigned and one where it isn't.